src/command_modules/azure-cli-interactive/azclishell/_dump_commands.py

```diff
@@ -127,13 +129,11 @@ def dump_command_table():
     with open(os.path.join(get_cache_dir(), command_file), 'w') as help_file:
         json.dump(data, help_file)
 
+    CMD_TABLE = cmd_table
 
-class Exporter(json.JSONEncoder):
-    def default(self, o):
-        try:
-            return super(Exporter, self).default(o)
-        except TypeError:
-            return str(o)
+
+def get_dumped_command_table():
+    return CMD_TABLE
```

Comment:
Why is CMD_TABLE global? Is there any reason you can't have multiple command tables (which you already do: one cached, one freshly generated) and then simply associate one of those instances with the interactive session?